js/services/Poll.js

```diff
@@ -6,14 +6,18 @@ define(['knockout', 'visibilityjs'], (ko, Visibility) => {
     isPageForeground(isForeground);
 
     if (isForeground) {
-      // when a user focuses tab, we shound immediately sync
+      // when a user focuses tab, we should immediately sync
       PollService.pollImmediately();
     }
   });
 
   class PollService {
+    constructor(){
+      this.isJobListMutated = ko.observable();
+      this.isJobListMutated.extend({ notify: 'always' });
+    }
     add(opts = {}, ...args) {
-      const { callback = () => {}, interval = 1000, isSilentAfterFirstCall = false } = opts;
+      const { callback = () => {}, interval = 1000, isSilentAfterFirstCall = false, isJobListMutated = false } = opts;
```

Comment:
I would name the `isJobListMutated` in a different way for 2 reasons:
1. It sounds like a statement regarding what happened / what is present, whereas in fact it says whether something should happen in future, i.e. whether the callback should trigger jobs list rerender
2. A bit difficult to distinguish it from the `this.isJobListMutated`

Reply:
I think this comment also applies to the `isJobListMutated` function in the `Poll.js`